### PR TITLE
First pass at introducing the Jobs table

### PR DESCRIPTION
**Pull request by pierlo-upitup**
This is the first iteration towards having a fully-functional Jobs table.

Current state:
jobs

To test, simply swap `JobsTab.js#L160` with 
`let item = DCOSStore.serviceTree.findItemById(id) || DCOSStore.serviceTree;`

As agreed with @orlandohohmeier who's working on the Structs, the `JobTree` is expected to mimic the `ServiceTree` structure and API. 

ACs

- [x] Render the Jobs table and allow browsing groups
- [x] Allow filtering the table by text
- [x] Persist search terms across page navigation with saveState 
- [x] Show the breadcrumb (but has known issues: see below)

Known issues
- [ ] the newly introduced `Breadcrumb` component does not seem to respond to route changes as seen in the gif
- [ ] the filters/breadcrumb headline swap has styling discrepancies. I believe @MatApple is going to show us the way here after refactoring of existing Breadcrumb components (such as in the Services page)
- [ ] the styles and text for the actual Job `status` and `lastRunStatus` are not currently rendered as in the mockups

For the last point, I'll wait until the actual Job struct is in place so that the API is agreed upon.

**NB** Tests will follow after first review iteration.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/components/JobSearchFilter.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import FilterInputText from './FilterInputText';
+import QueryParamsMixin from '../mixins/QueryParamsMixin';
+import JobFilterTypes from '../constants/JobFilterTypes';
+
+const METHODS_TO_BIND = ['setSearchString'];
+
+class JobSearchFilter extends mixin(QueryParamsMixin) {
+  constructor() {
+    super();
```
Comment (orlandohohmeier) [inline src/js/components/JobSearchFilter.js]: Please pass the arguments to the super class `super(...arguments);` as discussed in https://github.com/dcos/dcos-ui/pull/221#discussion_r63978069 and https://github.com/dcos/dcos-ui/pull/221#discussion_r64065922

File: src/js/stores/DCOSStore.js [hunk 2]
```diff
@@ -146,6 +148,12 @@ class DCOSStore extends EventEmitter {
 
   }
 
+  get jobTree() {
```
Comment (rcorral) [inline src/js/stores/DCOSStore.js]: I'm not sure it makes sense to couple Jobs into this struct.
Can we abstract common functionality and have separate structs?

Comment (orlandohohmeier) [inline src/js/stores/DCOSStore.js]: ?

Comment (pierlo-upitup) [inline src/js/stores/DCOSStore.js]: I think the Jobs belong here as much as the Services do. Or am I missing the point of having the DCOSStore?

**Comment by pierlo-upitup** [comment]
I've added a few tests basically replicating what the `ServiceTab-test.js` provided. Imho long as we don't have the `JobTree` and `Job` structs in place, these should be enough. 

Additional and more specific tests will be part of a later refactoring once the models are in place. This shouldn't hold merge for the time being.

File: src/js/pages/jobs/JobsTable.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    let heading = ResourceTableUtil.renderHeading(JobTableHeaderLabels);
+
+    return [
+      {
+        className,
+        headerClassName: className,
+        prop: 'name',
+        render: this.renderHeadline,
+        sortable: true,
+        sortFunction: ServiceTableUtil.propCompareFunctionFactory,
```
Comment (orlandohohmeier) [inline src/js/pages/jobs/JobsTable.js]: I'd like to either use a dedicated `JobsTable` util or have a unified util that works for both... Same is true for the `ResourceTableUtil`. What do you think?

File: src/js/pages/jobs/JobsTable.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  renderLastRunStatus(prop, job) {
+    if (job instanceof Tree) {
+      return null;
+    }
+
+    let className = 'text-mute';
+    let lastRunStatus = job.lastRunStatus || 'n/a';
+
+    if (lastRunStatus === 'Failed') {
+      className = 'text-danger';
```
Comment (mlunoe) [inline src/js/pages/jobs/JobsTable.js]: Could we use a classSet for this?

**Comment by pierlo-upitup** [comment]
Waiting on #246 to be merged.

**Comment by pierlo-upitup** [comment]
Update: as agreed with @rcorral the breadcrumb integration will be taken care of by @daiyiwhale

**Comment by pierlo-upitup** [comment]
UPDATE
------

I have refactored most of this PR to reflect the current state of the Chronos Store data.

As the Job struct PR (https://github.com/dcos/dcos-ui/pull/315) is not yet merged, I decided to stop at the most basic table implementation so that someone else can take over this view. 

What this PR achieves is the following:
image

To test, enable `useFixtures` in your `Config.dev.js`. 


cc @jfurrow @orlandohohmeier

**Comment by pierlo-upitup** [comment]
retest this please

**Comment by pierlo-upitup** [comment]
image

**Comment by pierlo-upitup** [comment]
retest this please

**Comment by pierlo-upitup** [comment]
Apologies for the self-merge – it was unintentional. We agreed with @orlandohohmeier that he's going to finish his review and if there are any problems with it I'll address them in a follow-up PR to prevent losing context with a reverted PR.

**Comment by orlandohohmeier** [comment]
I'm so happy that this finally landed. Great job, despite the following two issues 💯 ✨   which I'd like to see  addressed in a subsequent PR. What do you think?

**Jobs headline**
jobs-filter-heading mov
Let's always show the "Jobs" headline, similar to the "Services" headline.

**Table Utils**
Let's introduce dedicated table utils instead of reusing the service table utils here (e.g. `ServiceTableUtil.propCompareFunctionFactory`) are have general once as proposed in https://github.com/dcos/dcos-ui/pull/233#discussion_r64298163

**Comment by pierlo-upitup** [comment]
Thanks for the review!
1) the missing headline is due to the missing Breadcrumbs, which I know @daiyiwhale will take care of (see https://github.com/dcos/dcos-ui/pull/233/files#diff-ccdf865ebe848e0ea4ef4cc3c35450beR104) 
2) That makes sense. Let's do it in a separate PR

**Pull request merged**